### ISSUE TITLE
Allegro Punkty Kolporter przez API (wysyłam z allegro)

### ISSUE THREAD

**Issue by zomerstm** [allegro/allegro-api#4856]
Chciałbym nadać przesyłkę Allegro Punkty Kolporter. Wasi konsultanci kierują na opcję Wysyłam z Allegro. W jej dokumentacji, jest opisane, że potrzebuję serviceId, aby stworzyć paczkę przez wasze API. Niestety, jak skorzystam z usługi parcel-management/delivery-services, nie zwraca mi żadnej, tymbardziej tej usuługi. Jak mam nadać taką przesyłkę przez API?

**Comment by AureliuszBrussy** [comment]
Id usługi (serviceId) powinieneś pobrać właśnie dzięki [GET /parcel-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getAvailableDeliveryServices). Jaki kod odpowiedzi otrzymałeś? Proszę, prześlij w wątku swój pełen request (w curl) i trace-id, które zwróciliśmy w nagłówku respone'a.

**Comment by zomerstm** [comment]
No dobra, zgłupiałem. Możliwe, że klient coś skonfigurował od Soboty, czy coś się zsynchronizowało? Chciałem przygotować szczegóły i dostałem response, z opcjami dostawy, między innymi Allegro Punkty.
Rozumiem, że to jest ServiceID, którego mam użyć?
`[4] => Array
                (
                    [id] => 11365365
                    [service] => ALLEGRO
                    [name] => Allegro Punkty
                    [owner] => ALLEGRO
                    [carrierId] => ALLEGRO
                    [additionalServices] => Array
                        (
                            [cashOnDelivery] => Array
                                (
                                    [available] => 
                                    [expressAvailable] => 
                                )

                            [options] => Array
                                (
                                )

                        )

                )`

**Comment by AureliuszBrussy** [comment]
Tak, zgadza się, powinieneś skorzystać z id usługi o nazwie Allegro Punkty.

**Comment by zomerstm** [comment]
Ok, udało się przygotować dane do paczki, i ile walidację podczas dodawania przechodzę, po potem przy pobieraniu statusu otrzymuję błąd i nie jestem w stanie go obejść - chodzi o pickup.name (UUID: c434eb0f-cce2-4f56-b908-8ec95f7576bd). Dostaję błąd: Nazwa punktu nadania przesyłki jest wymagana. Wprowadziłem dane sklepu, do punktu wysyłki, bo tak naprawdę na początku będą te paczki dostarczane bezpośrednio do Kolportera. Próbowałem też te dane pominąć. Co tutaj powinno tak naprawdę być?

**Comment by AureliuszBrussy** [comment]
W pickup.name powinno znaleźć się imię i nazwisko nadawcy przesyłki. Powinno się tam znaleźć nawet, jeżeli adresem nadania będzie wybrany w pickup.pointId punkt.

**Comment by zomerstm** [comment]
Udało się. No jest napisane w dokumentacji person, ale trochę zgłupiałem, bo konto sklep. 
Jeszcze pytanie uzupełniające. W nagłówku, w przypadku przetwarzania powinien się pojawić parametr "Retry-After", akurat nie udalo mi się go uchwycić, ale oczywiście dodam wsparcie dla niego. Pytanie, jakich czasów się tutaj spodziewać, w kwestii generowania tych etykiet - nie wiem, czy kazać użytkownikowi czekać, czy jednak w tle będzie konieczne kontrolowanie pojawienia się etykiety?

**Comment by AureliuszBrussy** [comment]
To zależy od API poszczególnych przewoźników, zwykle jednak nie powinno to trwać dłużej niż kilkanaście sekund. W kwestii oczekiwania na potwierdzenie utworzenia przesyłki - być może któryś z użytkowników naszego forum podpowie, jak najlepiej to rozwiązać.

**Comment by zomerstm** [comment]
Ok, to już chyba finalne i ostateczne pytanie, bo paczka stworzona, etykieta pobrana. Podczas przygotowania paczki, przekazałem numer referencyjny, w items.description, ale na samej etykiecie nigdzie tego numeru niestety nie ma - wygląda jakby był nadpisany, informacjami od Allegro. Czy jest jakaś metoda przekazania czegokolwiek, aby było widoczne na etykiecie - łatwiej jest wtedy pracownikom magazynu obsługiwać takie etykiety.

**Comment by AureliuszBrussy** [comment]
Swoje id powinieneś przekazywać w polu **referenceNumber** - będzie ono widoczne na etykiecie.

**Comment by zomerstm** [comment]
Tam też je przekazuję, nawet specjalnie wywaliłem "/" bo z jakiś przyczyn je odrzucacie.

**Comment by AureliuszBrussy** [comment]
I nie widzisz na etykiecie tego numeru? Mogą wystąpić ograniczenia na znaki specjalne czy samą ilość znaków - zależnie od API przewoźnika.

**Comment by zomerstm** [comment]
No w label.referenceNumber dodałem numer "9017-08-21" i jest u was, bo jak pobieram szczegóły paczki, pokazuje go, ale na samej etykiecie, to albo jestem ślepy i klient też, albo nie ma, zresztą załączam: https://postimg.cc/qzTnns4b

**Comment by AureliuszBrussy** [comment]
Podaj proszę commandID pod którym utworzyłeś tą paczkę. Sprawdzimy to.

**Comment by zomerstm** [comment]
82ef756b-9157-4353-b21d-e9b392d91f2f

**Comment by AureliuszBrussy** [comment]
@zomerstm dla metody dostawy Allegro Punkty pole **label.referenceNumber** na etykiecie jest nadpisywane i aktualnie nie pokażemy przekazanej przez Ciebie wartości.
